include/aspect/introspection.h

```diff
@@ -51,23 +56,33 @@ namespace aspect
    * @ingroup Simulator
    */
   template <int dim>
-  struct Introspection
+  struct Introspection: public dealii::FEVariableSystem<dim>
```

Comment:
Yes, please. You constrain what we can and cannot do in deal.II if you step on another project's namespace. I'm very much in favor of putting this into the ASPECT namespace, simply because that's the code where it currently lives. I'm not opposed to eventually putting this into deal.II, just like we did with several functions from namespace Utilities, but in that case they also change namespace (as they need to if we want to stay backward compatible).